src/controller/CHIPDeviceController.cpp

```diff
@@ -1034,166 +1061,73 @@ void DeviceCommissioner::OnSessionEstablished()
 
     CommissioneeDeviceProxy * device = mDeviceBeingCommissioned;
 
     // PAI is being requested first - If PAI is not present, DAC will be requested next anyway.
     switch (mCertificateTypeBeingRequested)
     {
     case CertificateType::kDAC: {
-        device->SetDAC(certificate);
+        device->SetDAC(data.certificate);
         break;
     }
     case CertificateType::kPAI: {
-        device->SetPAI(certificate);
+        device->SetPAI(data.certificate);
         break;
     }
     case CertificateType::kUnknown:
     default: {
-        return CHIP_ERROR_INTERNAL;
+        OnSessionEstablishmentError(CHIP_ERROR_INCORRECT_STATE);
+        return;
     }
     }
 
     if (device->AreCredentialsAvailable())
     {
-        ChipLogProgress(Controller, "Sending Attestation Request to the device.");
-        ReturnErrorOnFailure(SendAttestationRequestCommand(device, device->GetAttestationNonce()));
+        StopSessionIf(SendAttestationRequestCommand(device, device->GetAttestationNonce()));
     }
     else
     {
-        CHIP_ERROR err = SendCertificateChainRequestCommand(device, CertificateType::kDAC);
-        if (err != CHIP_NO_ERROR)
-        {
-            ChipLogError(Controller, "Failed in sending Certificate Chain request command to the device: err %s", ErrorStr(err));
-            OnSessionEstablishmentError(err);
-            return err;
-        }
+        StopSessionIf(SendCertificateChainRequestCommand(device, CertificateType::kDAC));
     }
-
-    return CHIP_NO_ERROR;
 }
 
 CHIP_ERROR DeviceCommissioner::SendAttestationRequestCommand(CommissioneeDeviceProxy * device, const ByteSpan & attestationNonce)
 {
-    ChipLogDetail(Controller, "Sending Attestation request to %p device", device);
-    VerifyOrReturnError(device != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
-    chip::Controller::OperationalCredentialsCluster cluster;
-    cluster.Associate(device, 0);
-
-    Callback::Cancelable * successCallback = mAttestationResponseCallback.Cancel();
-    Callback::Cancelable * failureCallback = mOnAttestationFailureCallback.Cancel();
-
-    ReturnErrorOnFailure(cluster.AttestationRequest(successCallback, failureCallback, attestationNonce));
-    ChipLogDetail(Controller, "Sent Attestation request, waiting for the Attestation Information");
-    return CHIP_NO_ERROR;
-}
-
-void DeviceCommissioner::OnAttestationFailureResponse(void * context, uint8_t status)
-{
-    ChipLogProgress(Controller, "Device failed to receive the Attestation Information Response: 0x%02x", status);
-    DeviceCommissioner * commissioner = reinterpret_cast<DeviceCommissioner *>(context);
-    commissioner->mAttestationResponseCallback.Cancel();
-    commissioner->mOnAttestationFailureCallback.Cancel();
-    // TODO: Map error status to correct error code
-    commissioner->OnSessionEstablishmentError(CHIP_ERROR_INTERNAL);
+    OperationalCredentials::Commands::AttestationRequest::Type request;
+    request.attestationNonce = device->GetAttestationNonce();
```

Comment:
The code used to send `attestationNonce` here.  Why the change to `device->GetAttestationNonce()`?

Reply:
I have reverted it back.